### PR TITLE
Feat: Collect Module Follower-Only Toggles

### PR DESCRIPTION
**Pull request by Zer0dot**
This PR introduces the ability to choose whether you restrict a publication's collection to only followers, or anyone.

### CODE REVIEW

File: contracts/core/modules/collect/FeeCollectModule.sol [hunk 1]
```diff
@@ -61,21 +64,25 @@ contract FeeCollectModule is ICollectModule, FeeModuleBase, FollowValidationModu
...
             !_currencyWhitelisted(currency) ||
             recipient == address(0) ||
             referralFee > BPS_MAX ||
             amount == 0
         ) revert Errors.InitParamsInvalid();
 
-        _dataByPublicationByProfile[profileId][pubId].referralFee = referralFee;
+        _dataByPublicationByProfile[profileId][pubId].amount = amount;
         _dataByPublicationByProfile[profileId][pubId].recipient = recipient;
         _dataByPublicationByProfile[profileId][pubId].currency = currency;
```
Comment (donosonaumczuk) [inline contracts/core/modules/collect/FeeCollectModule.sol:83]: Generally I wouldn't mark this, but I saw you have been re-arranging those to match the order of the decoded data so maybe you would like to:

```suggestion
        _dataByPublicationByProfile[profileId][pubId].currency = currency;
        _dataByPublicationByProfile[profileId][pubId].recipient = recipient;
```

File: contracts/core/modules/collect/LimitedFeeCollectModule.sol [hunk 2]
```diff
@@ -81,9 +85,10 @@ contract LimitedFeeCollectModule is ICollectModule, FeeModuleBase, FollowValidat
 
         _dataByPublicationByProfile[profileId][pubId].collectLimit = collectLimit;
         _dataByPublicationByProfile[profileId][pubId].amount = amount;
-        _dataByPublicationByProfile[profileId][pubId].currency = currency;
         _dataByPublicationByProfile[profileId][pubId].recipient = recipient;
+        _dataByPublicationByProfile[profileId][pubId].currency = currency;
```
Comment (donosonaumczuk) [inline contracts/core/modules/collect/LimitedFeeCollectModule.sol:89]: Same here, was OK before the change

**Review by donosonaumczuk — CHANGES_REQUESTED** [review]
Code looks good, I commented two small things that probably you would like to change

File: contracts/core/modules/collect/LimitedFeeCollectModule.sol (hunk 2 above)
Comment (Zer0dot) [reply contracts/core/modules/collect/LimitedFeeCollectModule.sol]: Addressed in [03a17e5](https://github.com/aave/lens-protocol/pull/89/commits/03a17e55e3ee9ad1068ac3435a9c5b6074db2155) where I also changed the struct variable ordering to match!

File: contracts/core/modules/collect/FeeCollectModule.sol (hunk 1 above)
Comment (Zer0dot) [reply contracts/core/modules/collect/FeeCollectModule.sol]: Addressed in [03a17e5](https://github.com/aave/lens-protocol/pull/89/commits/03a17e55e3ee9ad1068ac3435a9c5b6074db2155) like the other.

**Review by donosonaumczuk — DISMISSED** [review]
Great, for me it's ready

**Review by donosonaumczuk — APPROVED** [review]
Looks great

**Pull request merged**